### PR TITLE
Added reasonable blocks for KMeans

### PR DESCRIPTION
**Pull request by amgrigoriev**
# Description
Please include a summary of the change. For large or complex changes please include enough information to introduce your change and explain motivation for it.

Changes proposed in this pull request:
-
-Added reasonable block size for KMeans
-

### CODE REVIEW

**Comment by amgrigoriev** [comment]
make test

**Comment by amgrigoriev** [comment]
http://intel-ci.intel.com/ea0b6d4c-3abd-f136-858a-8cdcd4b723a1?

File: algorithms/kernel/kmeans/oneapi/kmeans_dense_lloyd_batch_kernel_ucapi_impl.i [hunk 1]
```diff
@@ -127,7 +127,28 @@ Status KMeansDenseLloydBatchKernelUCAPI<algorithmFPType>::compute(const NumericT
...
+
+    if(gemmBlockSize < nMinRows)
+        return Status(ErrorKMeansNumberOfClustersIsTooLarge);
+
+    size_t datasetBlockSize = nValuesInBlock;
+    while(datasetBlockSize  > nValuesInBlock / nFeatures) {
+        datasetBlockSize >>= 1;
+    }
+
+    if(datasetBlockSize < nMinRows)
```
Comment (prutskov) [inline algorithms/kernel/kmeans/oneapi/kmeans_dense_lloyd_batch_kernel_ucapi_impl.i]: nit: use `if(expression) {...}`

File: algorithms/kernel/kmeans/oneapi/kmeans_dense_lloyd_batch_kernel_ucapi_impl.i [hunk 2]
```diff
@@ -127,7 +127,28 @@ Status KMeansDenseLloydBatchKernelUCAPI<algorithmFPType>::compute(const NumericT
...
-    const size_t blockSize = nRows;
+    const size_t nValuesInBlock = 1024 * 1024 * 1024 / sizeof(algorithmFPType);
+    const size_t nMinRows = 1;
+    size_t gemmBlockSize = nValuesInBlock;
+
+    while(gemmBlockSize  > nValuesInBlock / nClusters) {
+        gemmBlockSize >>= 1;
+    }
+
+    if(gemmBlockSize < nMinRows)
```
Comment (prutskov) [inline algorithms/kernel/kmeans/oneapi/kmeans_dense_lloyd_batch_kernel_ucapi_impl.i]: nit: use `if(expression) {...}`

File: algorithms/kernel/kmeans/oneapi/kmeans_dense_lloyd_batch_kernel_ucapi_impl.i [hunk 3]
```diff
@@ -127,7 +127,28 @@ Status KMeansDenseLloydBatchKernelUCAPI<algorithmFPType>::compute(const NumericT
...
+    size_t datasetBlockSize = nValuesInBlock;
+    while(datasetBlockSize  > nValuesInBlock / nFeatures) {
+        datasetBlockSize >>= 1;
+    }
+
+    if(datasetBlockSize < nMinRows)
+        return Status(ErrorIncorrectNumberOfFeatures);
+
+    size_t blockSize = datasetBlockSize > gemmBlockSize ? gemmBlockSize : datasetBlockSize;
+    if(blockSize > nRows)
```
Comment (prutskov) [inline algorithms/kernel/kmeans/oneapi/kmeans_dense_lloyd_batch_kernel_ucapi_impl.i]: nit: use `if(expression) {...}`

**Pull request merged**